### PR TITLE
Add "open in new tab" feature to Social Links Block

### LINKED ISSUE #20707: Social icons Block: Open link in a new tab.
It would be nice to have the possibility to open social icons in a new tab.

We also received a request for it here: https://wordpress.org/support/topic/social-icons-link-target-attributes/

Today the link attribute in Social Icons Block look like this:

Screen Shot 2020-03-08 at 19 15 00

@mkaz @jasmussen

### LINKED ISSUE #25455: Need add "open in new tab" feature in social share in Gutenberg
***@rkradadiya  noticed in the social share widget there is no option for the "open in new tab" feature.***
This has already been reported on __trac__ , as ticket [51250](https://core.trac.wordpress.org/ticket/51250)
🙂

### PR DESCRIPTION
**Pull request by fabiankaegy**
## Description

I have modified the `SocialLink` and `SocialLinks` Blocks to include the ability to open in new tab. This was implemented using the `contextProvider` API to pass the setting down to the children.

It fixes the issue outlined in #20707

Closes #20707

## How has this been tested?

This has been tested locally with by setting up a SocialLinks block and then updating it to make sure there are no deprecation errors. 

## Types of changes

New feature & Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by fabiankaegy** [comment]
@mkaz I just saw this ticket was a duplicate and saw your reply in one of the older PR's https://github.com/WordPress/gutenberg/pull/20740#issuecomment-597748635 I think with that we can close this aswell

**Comment by mkaz** [comment]
I was just reviewing this one and I'm not sure if the requiring only setting once is still worth holding out for, it is definitely the ideal scenario but something might be better than nothing. I'm not sure the efforts of allowing a parent block to pass down attributes to InnerBlock children.

**Review by mkaz — COMMENTED** [review]
I've tested this out and it works nicely.

@jasmussen we've had this before and overall, is it worth holding off the feature since it requires setting "open in new tab" on each link, or should we deploy. Ideal scenario is set once on parent and the children inherit.

**Comment by fabiankaegy** [comment]
@mkaz I took a stab at using the `contextProvider` option and updated this PR to include that implementation. We can always go back to the earlier version though it you prefer that :)

**Comment by fabiankaegy** [comment]
This implementation currently really only allows you to toggle `Open in new Tab` on or of. I have not added the text field to modify the `rel` attribute directly like it is done in the button or how I mimicked it in the earlier implementation.

**Comment by mkaz** [comment]
Much better, I hadn't used the context yet. I had earmarked the ticket for when it got complete, but hadn't circled back. I've got a minor change, but I think this is what we want :+1: 

I'll let Joen take a look and make sure he's happy with the change.

File: packages/block-library/src/social-links/edit.js
Comment (mkaz) [inline packages/block-library/src/social-links/edit.js:40]: Minor change to sentence case

```suggestion
						label={ __( 'Open links in new tab' ) }
```

File: packages/block-library/src/social-links/edit.js
Comment (mkaz) [inline packages/block-library/src/social-links/edit.js:38]: Minor change to sentence case:

```suggestion
				<PanelBody title={ __( 'Link settings' ) }>
```

**Review by mkaz — COMMENTED** [review]
Couple of minor changes.

Thanks again!

**Comment by jasmussen** [comment]
This is what I see:

Screenshot 2020-09-19 at 10 05 12

Screenshot 2020-09-19 at 10 05 29

Superbly awesome, works perfectly. Open in new tab should be a global toggle for all links inside, and off by default. This PR does that beautifully. If Marcus digs the code, ship this as fast as you can. Thank you!

**Review by mkaz — APPROVED** [review]
:+1:

**Comment by afercia** [comment]
On a post that already contained a Social Icons block with 6 links, I'm getting 6 PHP notices:

`Notice: Undefined index: openInNewTab in /srv/www/wordpress-develop/public_html/build/wp-content/plugins/gutenberg/build/block-library/blocks/social-link.php on line 18`

Screenshot 2020-09-24 at 14 39 34

@fabiankaegy @mkaz can you please have a look, when you have a chance?

**Comment by fabiankaegy** [comment]
@afercia That is odd I had that in my checks. Will go over it again and have a look. My bad. Thank you for finding this

**Pull request merged**
